### PR TITLE
83 Set Up Analytics

### LINKED ISSUE #83: Create analytics integration
## User Story

As a church I want to collect data and understand how users are using our product

#### Problem Details _(Optional)_

* Lets create a simple analytics package that can be tapped into to track app events, and funnel those events to configurable endpoints - can probably just port this: https://github.com/NewSpring/Apollos/blob/alpha/src/%40utils/analytics/index.js
* If a church doesn't want to use Sentry, all they have to do is replace the usage of sentry inside this one file
* I've been told Rock has an interactions table. Lets determine if it's something we should tap into (if so, lets create documentation or additional issues for doing so)

#### Acceptance Criteria _(Optional)_

* [ ] It should log data in sentry

### PR DESCRIPTION
**Pull request by richarddubay**
Fixes #83 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings (in app AND in storybook)
- [x] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer
- [x] Add a line to the changelog documenting the high-level change this PR introduces

## Reviewer

- [ ] Review CI build results
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review new stories if any apply
- [ ] Review test coverage: Are all new or changed states tested?
- [ ] Review new test logic
- [ ] Review the changelog entry

### CODE REVIEW

**Comment by richarddubay** [comment]
You can see the project and the errors it's already catching at https://sentry.io/newspring-church/apollos-project/

**Comment by richarddubay** [comment]
@conrad-vanl and @burkeshartsis I left a bunch of code in the file commented out. This was all the stuff that was being used for Amplitude (which we aren't implementing right this second). If it makes more sense I can remove it for now and we can get it back from NS Apollos when we need it. Up to you.

**Comment by richarddubay** [comment]
> Also, I think this probably needs the changes from #101 for the item.node.id errors to go away.

This has those changes now.

File: mobile/src/utils/analytics/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+const ScreenView = 'ScreenView';
+const Liked = 'Liked';
+const Shared = 'Shared';
+const Login = 'Login';
+const Register = 'Register';
+const ForgotPassword = 'ForgotPassword';
+const ContactedGroup = 'ContactedGroup';
+const GivingStarted = 'GivingStarted';
+const AudioPlayed = 'AudioPlayed';
+const AudioPaused = 'AudioPaused';
```
Comment (conrad-vanl) [inline mobile/src/utils/analytics/index.js]: Should we go ahead and remove Events that we aren't using (yet)?

File: mobile/src/utils/analytics/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  Login,
+  Register,
+  ForgotPassword,
+  ContactedGroup,
+  GivingStarted,
+  AudioPlayed,
+  AudioPaused,
+};
+
+// const nativeOnlyEvents = {
```
Comment (conrad-vanl) [inline mobile/src/utils/analytics/index.js]: Probably don't need this at all

File: mobile/src/utils/analytics/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+// export const identify = (userId, userProperties) => {
+//   if (Settings.NODE_ENV === 'development' || Settings.NODE_ENV === 'testing')
+//     return;
+//   instance.setUserId(userId);
+//   if (userProperties) instance.setUserProperties(userProperties);
+// };
+
+// export const trackScreen = (screenName, screenProperties) => {
+export const trackScreen = (screenName) => {
+  // track(events.ScreenView, {
```
Comment (conrad-vanl) [inline mobile/src/utils/analytics/index.js]: And reason why `trackScreen` is no longer calling `track`? I think the point of this was just for `trackScreen` to be shorthand for calling `track` with the `events.ScreenView` event

**Pull request merged**